app/controllers/your_claims_controller.rb

```diff
@@ -1,3 +1,10 @@
+require 'pagy/extras/array'
 class YourClaimsController < ApplicationController
-  def index; end
+  include Pagy::Backend
+  def index
+    claims = Claim.your_claims
+    claims = claims.map { |claim| BaseViewModel.build(:your_claims, claim) }
+    @pagy, @claims = pagy_array(claims)
+    @next_claim = Claim.unassigned_claims.order(created_at: :desc).first
```

Comment:
we should do this as if two please open the screen at the same time - wait 5 minutes, then click the link, they would both get the same claim. Add a new endpoint to select a claim, it will determine the new claim to assign, assign it and then redirect, this is a safer approach

Reply:
Created a separate ticket to discuss the best way to handle assigning next claim to current user. Please feel free to add any details to the ticket.

[https://dsdmoj.atlassian.net/browse/CRM457-577](https://dsdmoj.atlassian.net/browse/CRM457-577)